perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -54,6 +136,130 @@
                   'num_machines': 1}
 
 
+GLOBALS_TEMPLATE = """
+[global]
+ioengine=libaio
+invalidate=1
+direct=1
+runtime=10m
+time_based
+filename={{filename}}
+do_verify=0
+verify_fatal=0
+randrepeat=0
+"""
+
+JOB_TEMPLATE = """
+{% for iodepth in iodepths %}
+[{{name}}-io-depth-{{iodepth}}]
+stonewall
+rw={{rwkind}}
+blocksize={{blocksize}}
+iodepth={{iodepth}}
+size={{size}}
+{% endfor %}
+"""
+
+
+def GetIODepths(io_depths):
+  """Parse the io_depths parameter.
+
+  Args:
+    io_depths: a string in the format of the --io_depths flag.
+
+  Returns:
+    An iterable of integers.
+
+  Raises:
+    ValueError if io_depths doesn't follow a format it recognizes.
+  """
+
+  match = IODEPTHS_REGEXP.match(io_depths)
+
+  if match.group(2) is None:
+    return [int(match.group(1))]
+  else:
+    return range(int(match.group(1)), int(match.group(3)) + 1)
+
+
+def GenerateJobFileString(disk, against_device,
+                          scenarios, io_depths, working_set_size):
+  """Make a string with our fio job file.
+
+  Args:
+    disk: the disk.BaseDisk object we're benchmarking with.
+    against_device: bool. True if we're using a raw disk.
+    scenarios: iterable of dicts, taken from SCENARIOS.
+    io_depths: iterable. The IO queue depths to test.
+    working_set_size: int or None. If int, the size of the working set in GB.
+
+  Returns:
+    The contents of a fio job file, as a string.
+  """
+
+  if against_device:
+    filename = disk.GetDevicePath()
+  else:
+    filename = posixpath.join(disk.mount_point, DEFAULT_TEMP_FILE_NAME)
+
+  globals_template = jinja2.Template(GLOBALS_TEMPLATE,
+                                     undefined=jinja2.StrictUndefined)
+  job_template = jinja2.Template(JOB_TEMPLATE,
+                                 undefined=jinja2.StrictUndefined)
+
+  file_string = str(globals_template.render(filename=filename))
+  size_string = str(working_set_size) + 'G' if working_set_size else '100%'
+  for scenario in scenarios:
```

Comment:
Could you move all this into the template? Mixing logic between this function and the template seems confusing.
At least prefer building up a list and calling `str.join(iterable)` to repeatedly concatenating.